File: commands/branch-commands/switchBranch.js
import simpleGit from "simple-git";

const git = simpleGit();

export default {
  command: "switch-branch [branchName]",
  describe: "Switch to a branch",
  builder: (yargs) => {
    yargs.positional("branchName", {
      describe: "Name of the branch to switch to",
      type: "string",
      demandOption: true,
    });
  },
  handler: async (argv) => {
    try {
      await git.checkout(argv.branchName);
      console.log(`Switched to branch "${argv.branchName}".`);
    } catch (error) {
      console.error("Error switching branches:", error.message);
    }
  },
};

File: cmd/switchBranch.go
package cmd

import (
	"fmt"
	"log"
	"os/exec"

	"github.com/spf13/cobra"
) 

var SwitchBranchCmd = &cobra.Command{
	Use:   "switch",
	Short: "Switch to a different branch",
	Run: func(cmd *cobra.Command, args []string) {
		if len(args) < 1 {
			fmt.Println("Error: You must provide a branch name to switch to")
			return
		}

		branchName := args[0]

		err := exec.Command("git", "checkout", branchName).Run()
		if err != nil {
			log.Fatalf("Error switching to branch '%s': %v", branchName, err)
		}
		fmt.Printf("Switched to branch '%s' successfully\n", branchName)
	},
}

func init() {
	//No flags
}